test_rmw_implementation/test/test_publisher.cpp

```diff
@@ -527,6 +527,38 @@ TEST_F(CLASSNAME(TestPublisherUse, RMW_IMPLEMENTATION), publish_serialized_with_
   });
 }
 
+TEST_F(CLASSNAME(TestPublisherUse, RMW_IMPLEMENTATION), wait_for_all_acked_with_best_effort) {
+  test_msgs__msg__BasicTypes input_message{};
+  ASSERT_TRUE(test_msgs__msg__BasicTypes__init(&input_message));
+  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+  {
+    test_msgs__msg__BasicTypes__fini(&input_message);
+  });
+
+  rmw_subscription_options_t options = rmw_get_default_subscription_options();
+  rmw_subscription_t * sub = rmw_create_subscription(node, ts, topic_name, &qos_profile, &options);
+  ASSERT_NE(nullptr, sub) << rmw_get_error_string().str;
+  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+  {
+    rmw_ret_t ret = rmw_destroy_subscription(node, sub);
+    EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
+  });
+
+  size_t subscription_count = 0u;
+  rmw_ret_t ret = rmw_publisher_count_matched_subscriptions(pub, &subscription_count);
+  ASSERT_EQ(subscription_count, static_cast<size_t>(1));
+
+  ret = rmw_publish(pub, &input_message, nullptr);
+  ASSERT_EQ(ret, RMW_RET_OK);
+
```

Comment:
This isn't adding much to the test really, I would keep it simple:

```suggestion
```

Reply:
@ivanpauno Addressed your comments. https://github.com/ros2/rmw_implementation/pull/188/commits/87b997a7a46799a786fef5b1f0e81b491b58a70a